tests/test_smt.py

```diff
@@ -132,6 +132,52 @@ def test_get_cert_invalid_cert(mock_cert_pull, mock_logging):
     mock_logging.error.assert_called_with(msg)
 
 
+# ----------------------------------------------------------------------------
+@patch('smt.os.path.join')
+@patch('smt.logging')
+@patch('smt.requests.get')
+def test_write_cert_exception(
+    mock_request_get,
+    mock_logging,
+    mock_os_path_join
+):
+    """Handle exception when writing the cert and it failed fetching."""
+    mock_request_get.side_effect = Exception('FOO')
+    smt = SMT(etree.fromstring(smt_data_ipv46))
+    mock_os_path_join.return_value = '/tmp/registration_server_fc00__1.pem'
+    assert not smt.get_cert()
+    assert mock_logging.error.called
+    mock_logging.error.assert_called_with('Server 192.168.1.1 is unreachable')
+    assert not smt.write_cert('tmp')
+    os.remove('/tmp/registration_server_fc00__1.pem')
+    msg = 'write() argument must be str, not None'
+    mock_logging.error.assert_called_with(msg)
+
```

Comment:
This does not sufficiently test the `write_cert()` method. Otherwise a follow on problem from the fix for the original problem would have been noticed. The cert on a client that has IPv4 only but receives a config for IPv4 and Ipv6 update server gets witten with the IPv6 address. While ultimately immaterial for cert validation as the name doesn't matter it would break SUMa. When adding tests the effort to test comprehensively is relatively small. This fans out into 4 tests:

1. Update server with IPv4 only
2. Update server with IPv6 only
3. Update server with dual stack config
4. The exception path in `write_cert()`